packages/cardano-services/src/Http/HttpService.ts

```diff
@@ -56,7 +56,16 @@ export abstract class HttpService extends RunnableModule {
         return HttpServer.sendJSON(res, await handler(...args));
       } catch (error) {
         logger.error(error);
-        return HttpServer.sendJSON(res, new ProviderError(ProviderFailure.Unhealthy, error), 500);
+
+        let code = 500;
+        let providerError: ProviderError;
+
+        if (error instanceof ProviderError) {
+          providerError = error;
+          if (error.reason === ProviderFailure.NotFound) code = 404;
+        } else providerError = new ProviderError(ProviderFailure.Unhealthy, error);
+
+        return HttpServer.sendJSON(res, providerError, code);
```

Comment:
I think it's best to avoid using mutable variables - immutable style tends to be less error-prone.

Mutable variables have their use cases (sometimes just fits the problem well, or as performance optimization), but I don't think this is one of those cases.

```suggestion
        if (error instanceof ProviderError) {
          const code = error.reason === ProviderFailure.NotFound ? 404 : 500;
          return HttpServer.sendJSON(res, error, code);
        }
        const providerError = new ProviderError(ProviderFailure.Unhealthy, error);
        return HttpServer.sendJSON(res, providerError, 500);
```

Reply:
Hi @mkazlauskas ;

67efd3a71 should resolve this